applications/tari_dan_wallet_daemon/src/handlers/transaction.rs

```diff
@@ -479,12 +480,24 @@ pub async fn handle_publish_template(
 
     let fee_account = get_account_or_default(req.fee_account, &sdk.accounts_api())?;
 
+    // trying to optimize WASM binary
+    let wasm_binary = match WasmTemplateOptimizer::new().optimize(req.binary.as_slice()).await {
```

Comment:
nit: The WasmTemplateOptimizer struct is a glorified function. Up to you but I cant see a need now or in the future for some stateful struct for optimisation. 
```suggestion
    let wasm_binary = match optimize_wasm_template(req.binary.as_slice()).await {
```

Reply:
Yeah I was thinking the same, let me convert that as it could be still a struct if we would optimize for different purposes.